pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1533,4 +1535,49 @@ public boolean isBatchMessageCompatibleVersion() {
     public String getClientVersion() {
         return clientVersion;
     }
+
+    /**
+     * The transaction marker controller controls the commit marker or the abort marker publish to the specified topic.
+     * If it is a commit marker, we need to wait for the position it committed at the topic.
+     */
+    final static class TxnMarkerController implements Topic.PublishContext {
```

Comment:
it is unclear how this `PublishContext` is used for committing the transaction and how it interacts with transaction buffer.